api/src/main/java/org/cardanofoundation/rosetta/common/util/ParseConstructionUtil.java

```diff
@@ -454,68 +436,45 @@ public static Operation parseVoteMetadataToOperation(Long index, String transact
-            Networks.mainnet(), com.bloxbean.cardano.client.address.AddressType.Reward);
+    NetworkEnum network = NetworkEnum.fromProtocolMagic(networkProtocolMagic);
+    if (network != null) {
+      for (String owner : owners) {
+        Address address = CardanoAddressUtils.getAddress(
+                null,
+                HexUtil.decodeHexString(owner),
+                (byte) -32,
+                network.getNetwork(),
+                com.bloxbean.cardano.client.address.AddressType.Reward);
         poolOwners.add(address.getAddress());
       }
     }
     return poolOwners;
   }
 
-  public static String parsePoolRewardAccount(Integer network, PoolRegistration poolRegistration) {
+  public static String parsePoolRewardAccount(Long networkProtocolMagic, PoolRegistration poolRegistration) {
     String cutRewardAccount = poolRegistration.getRewardAccount();
-    if (poolRegistration.getRewardAccount().length() == 58) {
+    if (poolRegistration.getRewardAccount().length() == Constants.HEX_PREFIX_AND_REWARD_ACCOUNT_LENGTH) {
       cutRewardAccount = poolRegistration.getRewardAccount().substring(2);
     }
-    if (network == NetworkIdentifierType.CARDANO_TESTNET_NETWORK.getValue()) {
-      return CardanoAddressUtils.getAddress(null,
+    NetworkEnum networkEnum = NetworkEnum.fromProtocolMagic(networkProtocolMagic);
+    if (networkEnum != null) {
+      return CardanoAddressUtils.getAddress(
+              null,
               HexUtil.decodeHexString(cutRewardAccount),
               (byte) -32,
-              Networks.testnet(), com.bloxbean.cardano.client.address.AddressType.Reward)
-          .getAddress();
+              networkEnum.getNetwork(),
+              com.bloxbean.cardano.client.address.AddressType.Reward).getAddress();
     }
-    if (network == NetworkIdentifierType.CARDANO_PREPROD_NETWORK.getValue()) {
-      return CardanoAddressUtils.getAddress(null,
-              HexUtil.decodeHexString(cutRewardAccount),
-              (byte) -32,
-              Networks.preprod(), com.bloxbean.cardano.client.address.AddressType.Reward)
-          .getAddress();
-    }
-    if (network == NetworkIdentifierType.CARDANO_MAINNET_NETWORK.getValue()) {
-      return CardanoAddressUtils.getAddress(null,
-              HexUtil.decodeHexString(cutRewardAccount),
-              (byte) -32,
-              Networks.mainnet(), com.bloxbean.cardano.client.address.AddressType.Reward)
-          .getAddress();
-    }
-
     return null;
   }
 
   public static PoolMetadata parsePoolMetadata(PoolRegistration poolRegistration) {
     if (poolRegistration.getPoolMetadataUrl() != null
         || poolRegistration.getPoolMetadataHash() != null) {
-      return new PoolMetadata(poolRegistration.getPoolMetadataHash(),
-          poolRegistration.getPoolMetadataUrl());
+      return new PoolMetadata(poolRegistration.getPoolMetadataUrl(), poolRegistration.getPoolMetadataHash());
```

Comment:
Was it like this all along and working?

Reply:
looks like yes from 3/24/2024, there were no tests for this method